qiskit_experiments/visualization/__init__.py

```diff
@@ -0,0 +1,73 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+r"""
+=========================================================
+Visualization (:mod:`qiskit_experiments.visualization`)
+=========================================================
+
+.. currentmodule:: qiskit_experiments.visualization
+
+Visualization provides plotting functionality for creating figures from experiment and analysis results.
+This includes plotter and drawer classes to plot data in :py:class:`CurveAnalysis` and its subclasses.
+Plotters inherit from :class:`BasePlotter` and define a type of figure that may be generated from
+experiment or analysis data. For example, the results from :class:`CurveAnalysis` --- or any other
+experiment where results are plotted against a single parameter (i.e., :math:`x`) --- can be plotted
+using the :class:`CurvePlotter` class, which plots X-Y-like values.
+
+These plotter classes act as a bridge (from the common bridge pattern in software development) between
+analysis classes (or even users) and plotting backends such as Matplotlib. Drawers are the backends, with
+a common interface defined in :class:`BaseDrawer`. Though Matplotlib is the only officially supported
+plotting backend in Qiskit Experiments (i.e., through :class:`MplDrawer`), custom drawers can be
+implemented by users to use alternative backends. As long as the backend is a subclass of
+:class:`BaseDrawer`, and implements all the necessary functionality, all plotters should be able to
+generate figures with the alternative backend.
+
+To collate style parameters together, plotters and drawers store instances of the :class:`PlotStyle`
+class. These instances can be merged and updated, so that default styles can have their values
+overwritten.
+
+Plotter Library
+==============
+
+.. autosummary::
+    :toctree: ../stubs/
+    :template: autosummary/class.rst
+
+    BasePlotter
+    CurvePlotter
+
+Drawer Library
+==============
+
+.. autosummary::
+    :toctree: ../stubs/
+    :template: autosummary/class.rst
+
+    BaseDrawer
+    MplDrawer
+    LegacyCurveCompatDrawer
+
+Plotting Style
+==============
+
+.. autosummary::
+    :toctree: ../stubs/
+    :template: autosummary/class.rst
+
+    PlotStyle
+"""
+
+# PlotStyle is imported by .drawers and .plotters. Skip PlotStyle import for isort to prevent circular
+# import.
+from .style import PlotStyle  # isort:skip
```

Comment:
VSCode uses Pylints `[isort](https://pycqa.github.io/isort/)` program to sort imports, which also sorts alphabetically. This comment stopped `iosrt` from sorting this import, to prevent a circular import. However, I've managed to fix the circular import and thus removed the comment in 7c3ca90.